### PR TITLE
CHANGELOG: v2.0.0-beta.17

### PR DESCRIPTION
**Pull request by RingerJK**
### Description
CHANGELOG v2.0.0-beta.17

### CODE REVIEW

**Comment by RingerJK** [comment]
waiting for https://github.com/mapbox/mapbox-navigation-android/pull/4482

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,36 @@
...
 
+## Mapbox Navigation SDK 2.0.0-beta.17 - July 2, 2021
+
+For details on how v2 differs from v1 and guidance on migrating from v1 of the Mapbox Navigation SDK for Android to the v2 public preview, see [2.0 Navigation SDK Migration Guide](https://github.com/mapbox/mapbox-navigation-android/wiki/2.0-Navigation-SDK-Migration-Guide).
+
+### Changelog
+#### Features
+- Added options to the `RouteArrowOptions` class for customizing the scaling expressions used for maneuver arrows. [#4551](https://github.com/mapbox/mapbox-navigation-android/pull/4551)
+- Added new `HistoryEvent`: `HistoryEventPushHistoryRecord` allows read events set with **type**-**properties** structure. [#4567](https://github.com/mapbox/mapbox-navigation-android/pull/4567)
+- Added new api to request road shield asynchronously. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)
```
Comment (LukasPaczos) [inline CHANGELOG.md:13]:
```suggestion
- :warning: Added new APIs to request and render road shields asynchronously, see `MapboxManeuverApi#getRoadShields` and `MapboxManeuverView#renderManeuverShields` for details. Shields are not automatically downloaded anymore to allow for more granular and faster initial updates. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,36 @@
...
+- Exposed `RouteStepProgress#instructionIndex` that indicates which instruction out of the list of instructions for a step is the currently active one. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)
+
+#### Bug fixes and improvements
+- Fixed a bug where in some cases status updates were not generated after switching between offline and online states. [#4558](https://github.com/mapbox/mapbox-navigation-android/pull/4558)
+- Fixed an issue where `RouteOptions` weren't appended to onboard router results which could've caused incorrect reroutes that missed silent waypoints or other waypoint modifiers. [#4563](https://github.com/mapbox/mapbox-navigation-android/pull/4563)
+- Refactored dependency of `libnavui-resources` by changing from api to implementation. [#4568](https://github.com/mapbox/mapbox-navigation-android/pull/4568)
+- Added missing `IncidentInfo` properties: `countryCodeAlpha2`, `countryCodeAlpha3`, `lanesBlocked`, `longDescription`, `lanesClearDesc`, `numLanesBlocked`. [#4569](https://github.com/mapbox/mapbox-navigation-android/pull/4569)
```
Comment (LukasPaczos) [inline CHANGELOG.md:20]: This is a feature :tada:

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,36 @@
...
+- Refactored `MapboxManeuverApi` to return list of maneuvers asynchronously without the road shields. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)
```
Comment (LukasPaczos) [inline CHANGELOG.md:21]:
```suggestion
- :warning: Refactored `MapboxManeuverApi` to return a list of maneuvers asynchronously without the road shields and either for a raw route reference or for route progress. See `MapboxManeuverApi#getManeuvers` and `MapboxManeuverView#renderManeuvers` for details. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)
```

Comment (LukasPaczos) [reply CHANGELOG.md:21]: Let's place this item right above the road shields entry in features section.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,36 @@
...
+- Refactored road shield downloader to make it more efficient. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)
+- Changed `RouteProgress#bannerInstructions` does not become `null` and keeps its reference for as long as a step is active. Listen to `BannerInstructionsObserver` if you need a specific timing event when the banner first appears. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)
```
Comment (LukasPaczos) [inline CHANGELOG.md:23]:
```suggestion
- :warning: `RouteProgress#bannerInstructions` does not become `null` and keeps its reference for as long as a step is active. Listen to `BannerInstructionsObserver` if you need a specific timing event when the banner first appears. [#4482](https://github.com/mapbox/mapbox-navigation-android/pull/4482)
```

**Review by LukasPaczos — APPROVED** [review]

**Pull request merged**